File: OneDrive/Bureau/pi_rami/agro-ui/src/app/core/services/meteo.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { MeteoDTO, MeteoHistoriqueDTO } from '../models/meteo.model';

@Injectable({
  providedIn: 'root'
})
export class MeteoService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/meteo`;

  getMeteo(terrainId: number): Observable<MeteoDTO> {
    return this.http.get<MeteoDTO>(`${this.apiUrl}/${terrainId}`);
  }

  getHistorique(terrainId: number): Observable<MeteoHistoriqueDTO[]> {
    return this.http.get<MeteoHistoriqueDTO[]>(`${this.apiUrl}/historique/${terrainId}`);
  }
}

File: OneDrive/Bureau/pi_rami/Scoring_Aide/src/main/java/tn/esprit/scoringaideservice/service/MeteoService.java
package tn.esprit.scoringaideservice.service;

import tn.esprit.scoringaideservice.dto.MeteoDTO;
import java.util.List;
import tn.esprit.scoringaideservice.dto.MeteoHistoriqueDTO;

public interface MeteoService {

    MeteoDTO getMeteoByTerrain(Long terrainId);



    List<MeteoHistoriqueDTO> getHistorique(Long terrainId);

}
